src/screenComponents/hackingDialog.h

```diff
@@ -19,35 +21,25 @@ class GuiHackingDialog : public GuiOverlay
     void open(P<SpaceObject> target);
     virtual void onDraw(sf::RenderTarget& window) override;
     virtual bool onMouseDown(sf::Vector2f position) override;
+    void onMiniGameComplete(bool success);
+
 private:
     P<SpaceObject> target;
     string target_system;
-    
-    static constexpr int hacking_field_size = 8;
-    static constexpr int bomb_count = 8;
+    float reset_time;
     static constexpr float auto_reset_time = 2.0f;
-    GuiPanel* minigame_box;
+    bool last_game_success;
     GuiLabel* status_label;
     GuiLabel* hacking_status_label;
     GuiButton* reset_button;
+    GuiButton* close_button;
     GuiProgressbar* progress_bar;
-    int error_count;
-    int correct_count;
-    float reset_time;
-    class FieldItem
-    {
-    public:
-        GuiToggleButton* button;
-        bool bomb;
-    };
-    FieldItem field_item[hacking_field_size][hacking_field_size];
-    
+
+    GuiPanel* minigame_box;
+    MiniGame* game;
```

Comment:
If you make the minigame a std::shared_ptr, you don't need to keep track of deleting it, and these is then no risk of memory leaks:
https://en.cppreference.com/w/cpp/memory/shared_ptr

(this isn't needed for the gui elements, as they manage their own lifetime and get deleted with their parent)

Reply:
Agreed. Done.